pkg/api/types.go

```diff
@@ -12,6 +12,30 @@ type ImageTag struct {
 	Timestamp    time.Time    `json:"timestamp"`
 	OS           OS           `json:"os,omitempty"`
 	Architecture Architecture `json:"architecture,omitempty"`
+
+	// If this is a Manifest list we need to keep them together
+	Children []*ImageTag `json:"children,omitempty"`
+}
+
+func (i *ImageTag) HasChildren() bool {
+	return len(i.Children) > 0
+}
+
+func (i *ImageTag) MatchesSHA(sha string) bool {
+	if sha == i.SHA {
+		return true
+	}
+	for _, known := range i.Children {
+		if known.MatchesSHA(sha) {
```

Comment:
Possible to set up an infinite loop here with circular references.

I don't think you'd ever need/want to go more than one level deep, so you could do `if known.SHA == sha` instead, which would be safer.